asab/library/service.py

```diff
@@ -514,7 +505,50 @@ def on_library_change(self, message, provider, path):
 		if isinstance(paths, str):
 			paths = [paths]
 		for path in paths:
-			assert path[:1] == '/', "Absolute path must be used when subscribing to the library changes"
+			if not path.startswith("/"):
+				raise LibraryInvalidPathError(
+					message="Directory path must start with '/' when subscribing to Library changes.",
+					path=path,
+				)
 
 			for provider in self.Libraries:
 				await provider.subscribe(path)
+
+
+def _validate_path_item(path: str) -> None:
```

Comment:
You can also add these special checks to validate that paths do not contain ~  or .. , which could lead to security issues or unintended behaviour for both directory and item paths.

Reply:
Done :)